docs/reference/registry/registry-permissions.md

```diff
@@ -0,0 +1,45 @@
+| GetStreamFeatureView     | StreamFeatureView   | Read                   | Get a stream feature view from the registry     |
+| ListStreamFeatureViews   | StreamFeatureView   | Read                   | List stream feature views in the registry       |
+| GetOnDemandFeatureView   | OnDemandFeatureView | Read                   | Get an on-demand feature view from the registry |
+| ListOnDemandFeatureViews | OnDemandFeatureView | Read                   | List on-demand feature views in the registry    |
+| ApplyFeatureService      | FeatureService      | Create, Update, Delete | Apply a feature service to the registry         |
+| GetFeatureService        | FeatureService      | Read                   | Get a feature service from the registry         |
+| ListFeatureServices      | FeatureService      | Read                   | List feature services in the registry           |
+| DeleteFeatureService     | FeatureService      | Delete                 | Delete a feature service from the registry      |
+| ApplySavedDataset        | SavedDataset        | Create, Update, Delete | Apply a saved dataset to the registry           |
+| GetSavedDataset          | SavedDataset        | Read                   | Get a saved dataset from the registry           |
+| ListSavedDatasets        | SavedDataset        | Read                   | List saved datasets in the registry             |
+| DeleteSavedDataset       | SavedDataset        | Delete                 | Delete a saved dataset from the registry        |
+| ApplyValidationReference | ValidationReference | Create, Update, Delete | Apply a validation reference to the registry    |
+| GetValidationReference   | ValidationReference | Read                   | Get a validation reference from the registry    |
+| ListValidationReferences | ValidationReference | Read                   | List validation references in the registry      |
+| DeleteValidationReference| ValidationReference | Delete                 | Delete a validation reference from the registry |
+| ApplyPermission          | Permission          | Create, Update, Delete | Apply a permission to the registry           |
+| GetPermission            | Permission          | Read                   | Get a permission from the registry           |
+| ListPermissions          | Permission          | Read                   | List permissions in the registry             |
+| DeletePermission         | Permission          | Delete                 | Delete a permission from the registry        |
+| Commit                   |                     | None                   | Commit changes to the registry               |
+| Refresh                  |                     | None                   | Refresh the registry                         |
+| Proto                    |                     | None                   | Get the proto representation of the registry |
```

Comment:
@tokoko any thoughts how to `assert_permissions` on this missing function.

Reply:
for `Proto`, which is basically "read everything" endpoint, we should peek inside filter out only the objects that the user has `read` permission on. It will be a bit awkward and slow unfortunately as `proto` method returns protobuf object rather than feast python objects, each of which I'm assuming we will have to first convert to python before applying permissions and returning "reduced" proto object back to the user. @dmartinol

btw, this endpoint is used for client-side caching so this will be by far the most commonly called endpoint on the server.

Reply:
> for Proto, which is basically "read everything" endpoint, we should peek inside and filter out only the objects that the user has read permission on.

Agree, but only when it comes from a direct client request to the `Proto` endpoint.
When the request comes from another service it should be unfiltered and return the full content instead.

BTW: how to allow this service-2-service communication is still to be finalized. We have a solution based on a sort of `super-role` that we automatically create (k8s demo), but it's not optimized as it runs unnecessary validations at every requests. Service-2-service requests should be easily identified and promptly permitted without additional overhead.

Reply:
> BTW: how to allow this service-2-service communication is still to be finalized. We have a solution based on a sort of `super-role` that we automatically create (k8s demo), but it's not optimized as it runs unnecessary validations at every requests. Service-2-service requests should be easily identified and promptly permitted without additional overhead.

I was also thinking of some sort of built-in magic `admin` role. Instead of adding a `Permission` object for it, we can simply check for `admin` role right at the start of `assert_*` methods and sidestep the whole thing if the user is admin w/o any validation.

Reply:
How about we delay Commit, Refresh, Proto handling for a separate PR? This one is large enough as is.

Reply:
> I was also thinking of some sort of built-in magic `admin` role. Instead of adding a `Permission` object for it, we can simply check for `admin` role right at the start of `assert_*` methods and sidestep the whole thing if the user is admin w/o any validation.

This is one of the options under investigation. We want to mark the FS as a "server" when it starts one of the servers, so that each remote proxy, instead of generating the token from the auth provider, it sends something special (*) that the receiver can identify as a server request and skip the authorization enforcements steps.  
(*) under discussion: header, hashed code ...

Reply:
I'll resolve this and create a separate ticket